### PR TITLE
[RLlib] Training step docs

### PR DESCRIPTION
**Pull request by avnishn**
Signed-off-by: Avnish <avnishnarayan@gmail.com>

## Why are these changes needed?
This is documentation for the training step function that needs to also be cherry picked on the RC1 2.0 branch

https://github.com/ray-project/ray/pull/25432#pullrequestreview-1004265070

This change was actually previously approved by @sven1977

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by avnishn** [comment]
This was reviewed once before and merged so you can safely not look to carefully if in a rush for review

**Review by gjoliver — DISMISSED** [review]

**Comment by richardliaw** [comment]
PR description please? what's the reason for revert?

**Comment by gjoliver** [comment]
actually the title of this PR should not be "Revert XXX", but "Merge Training Step docs"

**Comment by avnishn** [comment]
https://github.com/ray-project/ray/pull/25432#pullrequestreview-1004265070

This branch was actually previously approved by @sven1977

File: doc/source/rllib/core-concepts.rst [hunk 1]
```diff
@@ -85,13 +85,13 @@ which implements the proximal policy optimization algorithm in RLlib.
...
-        from ray.rllib.algorithms.ppo import PPO, PPOConfig
-        config = PPOConfig().environment(env="CartPole-v0").training(train_batch_size=4000)
+        from ray.rllib.algorithms import PPOConfig
+        config = PPOConfig().environment("CartPole-v0").training(train_batch_size=4000)
 
         # Train via Ray Tune.
         # Note that Ray Tune does not yet support AlgorithmConfig objects, hence
         # we need to convert back to old-style config dicts.
-        tune.Tuner(PPO, param_space=config.to_dict()).fit()
+        tune.run("PPO", config=config.to_dict())
```
Comment (sven1977) [inline doc/source/rllib/core-concepts.rst:94]: Should we use the new `tune.Tuner()` here?

**Review by sven1977 — APPROVED** [review]
LGTM. Thanks for making progress on the `training_step` docs @avnishn !
Really cool.

**Pull request merged**